packages/seal/src/Schema/Index.php

```diff
@@ -58,6 +60,23 @@ public function __construct(
         $this->identifierField = $attributes['identifierField'];
     }
 
+    public function withDistinctAttribute(string|null $distinctAttribute): self
+    {
+        if (null !== $distinctAttribute && !\in_array($distinctAttribute, $this->filterableFields, true)) {
+            throw new \LogicException('The distinct attribute has to be filterable.');
+        }
+
+        $clone = clone $this;
+        $clone->distinctAttribute = $distinctAttribute;
+
+        return $clone;
+    }
```

Comment:
would not add wither methods here we should go the same way as for searchable, filterable, sortable, ... 

I added a deeper description here how I would add such case: https://github.com/PHP-CMSIG/search/issues/550#issuecomment-2910279969

PS: its in SEAL responsible if a field is defined as distinctable and the engine requires it to be also filterable that the schema manager takes core of that.

Reply:
But you cannot have more than one distinct field.

Reply:
I mean, technically you could but in Algolia it has to be known when creating the index.